services-api/src/main/java/io/scalecube/services/ServiceCall.java

```diff
@@ -487,7 +484,10 @@ private ServiceUnavailableException noReachableMemberException(ServiceMessage re
   }
 
   private Function<ServiceMessage, Object> toResponse() {
-    return message -> message.hasData() ? message.data() : UNEXPECTED_EMPTY_RESPONSE;
+    return message ->
+        message.hasData()
+            ? message.data()
+            : ServiceMessage.error(message.qualifier(), 503, 503, "Unexpected empty response");
```

Comment:
Why this method exists et al? I understand that neither you (and me) created it, but do u remember by chance?